src/sentry/models/groupsubscription.py

```diff
@@ -188,6 +189,13 @@ class GroupSubscription(Model):
     class Meta:
         app_label = "sentry"
         db_table = "sentry_groupsubscription"
-        unique_together = (("group", "user_id"),)
+        unique_together = (("group", "user_id"), ("group", "team_id"))
+        constraints = [
+            models.CheckConstraint(
+                check=models.Q(team_id__isnull=False, user_id__isnull=True)
+                | models.Q(team_id__isnull=True, user_id__isnull=False),
+                name="subscription_team_or_user_check",
+            )
+        ]
```

Comment:
```suggestion
        unique_together = (("group", "user_id"), ("group", "team"))
        constraints = [
            models.CheckConstraint(
                check=models.Q(team_id__isnull=False, user_id__isnull=True)
                | models.Q(team_id__isnull=True, user_id__isnull=False),
                name="subscription_team_or_user_check",
            )
        ]
```

Reply:
gotcha @markstory so in the `models.CheckConstraint`, should the check still be kept as `team_id__isnull` or should it be changed to `team__isnull`?

Reply:
nevermind, i see in other parts of the code, the check should be kept, thanks!

Reply:
When you create a `FlexibleForeignKey` field Django makes both a relationship property that has the related object, and a `{relation}_id` field. Both are available in the model and can be used in constraints.